packages/astro/src/core/config/config.ts

```diff
@@ -226,54 +220,24 @@ export async function openConfig(configOptions: LoadConfigOptions): Promise<Open
 	};
 }
 
-interface TryLoadConfigResult {
-	value: Record<string, any>;
-	filePath?: string;
-}
-
 async function tryLoadConfig(
 	configOptions: LoadConfigOptions,
 	root: string
-): Promise<TryLoadConfigResult | undefined> {
+): Promise<Record<string, any> | undefined> {
```

Comment:
While we're refactoring, I want to propose the following change, with two options:
1. rename `tryLoadConfig` to `loadConfig` and always return a `Record<string, any>`. The function will return `{}` if no `configPath` is returned;
2. keep the name `tryLoadConfig` but throw a runtime error, and let the consumer handle the error;

What do you think?

It's a pity that there's no way to signal that a function can throw an error to a consumer :(

Reply:
Good call! I didn't spot the `loadConfig` opportunity (option 1) which we can definitely do. Went with that instead since throwing errors are a bit expensive.